### DISCUSSION TITLE
keymap conflict with lsp how to disable the lsp keymap?

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#3119] [Q&A] [answered]
I have a keymap `keymap("x", "<leader>cc", '"+y', { silent = true, noremap = true })` in the keymaps.lua file. 

But is not stable sometime it will be mapped to the lsp keymap (which is a lazyvim built-in map). So how to fix it? 

The lsp map is `{ "<leader>cc", vim.lsp.codelens.run, desc = "Run Codelens", mode = { "n", "v" }, has = "codeLens" },`

**Comment by dpetka2001** [comment]
Don't the [docs](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) already answer that? What is confusing?

**Reply by OriLiMu** [reply]
I add the code below to file `nvim-lspconfig.lua` in the plugin folder not worked. 

```lua
return {
  "neovim/nvim-lspconfig",
  init = function()
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    -- change a keymap
    keys[#keys + 1] = { "<leader>cc", '"+y' }
  end,
}

```

**Reply by dpetka2001** [reply, marked as answer]
In lspconfig just do `keys[#keys + 1] = { "<leader>cc", false }` and then define your own keymap in `/lua/config/keymaps.lua`.

**Reply by OriLiMu** [reply]
Solved I forget to set the mode. But seems something wrong with the defination of mode.
In my case, I only use this keymap in visual mode, so mode should be "x". But "x" not work for me. I have to make the mode to "v". It works. Why? 

About the mode: https://vi.stackexchange.com/questions/38859/what-does-mode-x-mean-in-neovim. 

And thanks for you help. 

```lua
return {
  "neovim/nvim-lspconfig",
  init = function()
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    keys[#keys + 1] = { "<leader>cc", '"+y', mode = "v" }
  end,
}
```

**Reply by dpetka2001** [reply]
For me it works with `x` when I select text with `v`+text_object_motion (for example I do `viw` to select a word and then `<leader>cc` and it yanks the word).

**Reply by OriLiMu** [reply]
hi the keymap works in file with certain ext (like: lua, py ... sth) . 
But it wont work in the `a.txt` or `.log` file. How to fix it?

**Reply by dpetka2001** [reply]
Because you defined the keymap inside the `init` function of `nvim-lspconfig`. That means that the keymap is only available when a LSP server is attached to the buffer. That's why I told you to just disable the keymap in `nvim-lspconfig` and then define your own in `/lua/config/keymaps.lua`.
